### PR TITLE
[Method References 2] LPS-166968 Convert method reference to ServiceTrackerMap in module data-engine-taglib and data-engine-rest-impl

### PR DESCRIPTION
**Pull request by W2jx**
Previous: https://github.com/liferay-core-infra/liferay-portal/pull/2915 and https://github.com/liferay-core-infra/liferay-portal/pull/2553

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:data-engine

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
no conflicts(2/22 8:22)

**Comment by W2jx** [comment]
no conflicts(2/27 8:30)

**Comment by W2jx** [comment]
no conflicts(2/28 9:42)

**Comment by W2jx** [comment]
no conflicts(3/1 10:45)

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/content/type/DataDefinitionContentTypeRegistryImpl.java [hunk 1]
```diff
@@ -37,21 +37,25 @@
...
 	@Override
 	public Long getClassNameId(String contentType) {
-		Long id = _classNameIds.get(contentType);
-
-		if (id == null) {
-			throw new DataDefinitionValidationException.MustSetValidContentType(
-				contentType);
-		}
+		DataDefinitionContentType dataDefinitionContentType =
+			getDataDefinitionContentType(contentType);
```
Comment (tinatian) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/content/type/DataDefinitionContentTypeRegistryImpl.java:39]: @W2jx Why you skip the null checking?

**Comment by tinatian** [comment]
@W2jx For https://github.com/liferay-core-infra/liferay-portal/pull/3575/commits/0f642f88130fd6408b746686cb0535d8b79397e7, To fix it, please separate population of model resources from this class.

**Comment by tinatian** [comment]
@W2jx Please always pay more attention to a commit marked with SF, it should not have logic changes. https://github.com/liferay-core-infra/liferay-portal/pull/3575/commits/d2d80f62d85787bba3957940ed5a723ccde5b3f0

**Comment by tinatian** [comment]
@W2jx This is not a simple convert from method reference to field reference, you are applying a new API (DataDefinitionContentTypeRegistry)  https://github.com/liferay-core-infra/liferay-portal/pull/3575/commits/0fadfdb31cd84bf20089297a00ed2053e3f2fe72.

**Comment by tinatian** [comment]
@W2jx Please see my comments and resend.

Thanks.
Tina

**Pull request closed without merging**